src/world_model/simple_gridworld.py

```diff
@@ -477,6 +481,13 @@ def transition_distribution(self, state, action, n_samples = None) -> dict:
             if imm_states[self.immobile_object_indices[loc]] == 0:
                 # turn state to 1:
                 imm_states = set_entry(imm_states, self.immobile_object_indices[loc], 1)
+        elif cell_type == 'd':
+            # flip switch and door states
+            previous_switch_state = imm_states[self.immobile_object_indices[loc]]
+            imm_states = set_entry(imm_states, self.immobile_object_indices[loc], (previous_switch_state + 1) % 2)
+            door_index = self.immobile_object_types.index('D')
```

Comment:
this looks good so far! in future, if we wanted to have multiple doors/switch pairs, would it be possible to implement that too?

Reply:
I think so, for example by numbering the doors and switches. However, as a result, we would need multiple characters to specify a door or a switch (e.g. 'D1', 'd1'). After talking about this possibility with Jobst, he said this likely wouldn't be a problem if we used the matrix notation for a grid world as opposed to defining grid worlds using .json files.